Original patch from gentoo: gentoo/src/patchsets/gcc/4.3.1/gentoo/11_all_gcc-netbsd-symbolic.patch
http://bugs.gentoo.org/122698

diff -durN gcc-4.3.1.orig/gcc/config/netbsd-elf.h gcc-4.3.1/gcc/config/netbsd-elf.h
--- gcc-4.3.1.orig/gcc/config/netbsd-elf.h	2007-09-03 18:14:04.000000000 +0200
+++ gcc-4.3.1/gcc/config/netbsd-elf.h	2008-06-10 14:57:48.000000000 +0200
@@ -82,6 +82,7 @@
 #define NETBSD_LINK_SPEC_ELF \
   "%{assert*} %{R*} %{rpath*} \
    %{shared:-shared} \
+   %{symbolic:-Bsymbolic} \
    %{!shared: \
      -dc -dp \
      %{!nostdlib: \
